lib/pbench/test/unit/agent/task/test_copy_result_tb.py

```diff
@@ -147,20 +155,25 @@ def test_connection_error(self, monkeypatch, agent_logger):
             responses.PUT, upload_url, body=requests.exceptions.ConnectionError("uh-oh")
         )
 
-        monkeypatch.setattr(Path, "exists", self.get_path_exists_mock(tb_name, True))
-        monkeypatch.setattr(
-            Path, "open", self.get_path_open_mock(tb_name, io.StringIO(tb_contents))
-        )
-
-        with pytest.raises(RuntimeError) as excinfo:
-            crt = CopyResultTb(
-                tb_name,
-                len(tb_contents),
-                "someMD5",
-                self.config,
-                agent_logger,
+        with monkeypatch.context() as m:
+            m.setattr(Path, "exists", self.get_path_exists_mock(tb_name, True))
+            m.setattr(
+                Path, "open", self.get_path_open_mock(tb_name, io.StringIO(tb_contents))
             )
-            crt.copy_result_tb("token")
+
+            with pytest.raises(RuntimeError) as excinfo:
+                crt = CopyResultTb(
+                    tb_name,
+                    len(tb_contents),
+                    "someMD5",
+                    self.config,
+                    agent_logger,
+                )
+                res = crt.copy_result_tb("token")
+                # we should never get here but flake8 flakes out
+                # complaining about the unused variable, so we
+                # might as well use it.
+                assert not res.ok
```

Comment:
Or just drop the `res =`??? If `not res.ok` I think we expect to see an exception, so this path should be more like an `assert False` ... but that's implicit in the `with pytest.raises`.